### PR TITLE
Add Service Lifecycle Feature Doc

### PR DESCRIPTION
**Pull request by agunde406**

### CODE REVIEW

File: community/planning/service_lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    }
+}
+```
+
+There are 4 types of commands that can be used to update a service. A service
+can be prepared, finalized, retired or purged. The first 3 commands result in an
+updated status in the `Lifecycle` store along with any operations that are
+specific for the service type, while purge should result in all service
+information being removed. The `Lifecycle` will return a command that will be
+combined with the lifecycle command and executor within the same transaction.
```
Comment (isabeltomb) [inline community/planning/service_lifecycle.md:108]: s/executor/executed

File: community/planning/service_lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        command_generator: LifecycleCommandGenerator<E::Context>,
+        command_executor: E,
+    ) -> Result<LifecycleExecutor<E>, InternalError> {
+      // omitted for brevity
+    }
+}
+```
+
+All commands, both service specific and lifecycle, will be atomically executed
+by a `StoreCommandGenerator`.
```
Comment (isabeltomb) [inline community/planning/service_lifecycle.md:136]: s/StoreCommandGenerator/StoreCommandExecutor

File: community/planning/service_lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+![Lifecycle Integration Admin Service]({% link images/admin_lifecycle.svg %}
+  "Lifecycle Integration with Admin Service"){:.centered}
+
+Due to the difference in designs, two of the required methods on the
+`LifecycleDispatch` are NO-OPs for the `SyncLifecycleInterface`. Services do not
+need to be stopped and restarted on shutdown. Therefore `shutdown_all_services`
+and `add_stopped_service` simply return `Ok(())`, same for if `add_service` is
+called and if the service was already added.
+
+`SyncLifecycleInterface` combines both `prepare_service` and `finish_service` to
```
Comment (isabeltomb) [inline community/planning/service_lifecycle.md:255]: is `finish_service` supposed to be `finalize_service`?

File: community/planning/service_lifecycle.md (hunk 1 above)
Comment (agunde406) [reply community/planning/service_lifecycle.md]: fixed

File: community/planning/service_lifecycle.md (hunk 2 above)
Comment (agunde406) [reply community/planning/service_lifecycle.md]: fixed

File: community/planning/service_lifecycle.md (hunk 3 above)
Comment (agunde406) [reply community/planning/service_lifecycle.md]: fixed

**Review by isabeltomb — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**